templates/kotlin/src/main/java/io/appwrite/Client.kt.twig

```diff
@@ -0,0 +1,232 @@
+            .build()
+
+        val httpBuilder = (endPoint + path).toHttpUrl().newBuilder()
+
+        if ("GET" == method) {
+            params.forEach {
+                when (it.value) {
+                    null -> {
+                        return@forEach
+                    }
+                    is List<*> -> {
+                        httpBuilder.addQueryParameter(it.key + "[]", it.value.toString())
+                    }
+                    else -> {
+                        httpBuilder.addQueryParameter(it.key, it.value.toString())
+                    }
+                }
+            }
+            val request = Request.Builder()
+                .url(httpBuilder.build())
+                .headers(requestHeaders)
+                .get()
+                .build()
+
+            return awaitResponse(request)
+        }
+
+        val body = if (MultipartBody.FORM.toString() == headers["content-type"]) {
+            val builder = MultipartBody.Builder().setType(MultipartBody.FORM)
+
+            params.forEach {
+                when {
+                    it.key == "file" -> {
+                        val file = it.value as File
+                        builder.addFormDataPart(it.key, file.name, file.asRequestBody())
+                    }
+                    it.value is List<*> -> {
+                        val list = it.value as List<*>
+                        for (index in list.indices) {
+                            builder.addFormDataPart(
+                                "${it.key}[]",
+                                list[index].toString()
+                            )
+                        }
+                    }
+                    else -> {
+                        builder.addFormDataPart(it.key, it.value.toString())
+                    }
+                }
+            }
+            builder.build()
+        } else {
+            Gson().toJson(params)
+                .toRequestBody("application/json".toMediaType())
+        }
+
+        val request = Request.Builder()
+            .url(httpBuilder.build())
+            .headers(requestHeaders)
+            .method(method, body)
+            .build()
+
+        return awaitResponse(request)
+    }
+
+    @Throws({{ spec.title | caseUcfirst }}Exception::class)
+    private suspend fun awaitResponse(request: Request) = suspendCancellableCoroutine<Response> {
+        http.newCall(request).enqueue(object : Callback {
+            override fun onFailure(call: Call, e: IOException) {
+                if (it.isCancelled) {
+                    return
+                }
+                it.resumeWithException(e)
+            }
+
+            override fun onResponse(call: Call, response: Response) {
+                response.use { res ->
+                    if (res.code >= 400) {
+                        val err = response.body?.string()?.fromJson(Error::class.java)
+                        throw AppwriteException("${err?.code}: ${err?.message}")
```

Comment:
Okay, I got the issue. Instead of throwing the exception here, you need to cancel the job with the `cause` 
like so 
```suggestion
                        it.cancel(AppwriteException("${err?.code}: ${err?.message}"))
```